onnx/shape_inference/implementation.h

```diff
@@ -12,6 +12,37 @@
 namespace ONNX_NAMESPACE {
 namespace shape_inference {
 
+struct SymbolTable {
+  SymbolTable(GraphProto* g) : index_(0){
```

Comment:
GraphProto& g?
if you use a pointer then you should check for nullptr... if nullptr is unacceptable then you should simply use a reference and make this a caller issue...